### PR TITLE
fix(delay): accepts absolute time delay

### PR DESCRIPTION
**Pull request by kwonoj**
- delay operator accepts absolute time as well as relative time
- add additional test coverages

relates to #549

### CODE REVIEW

File: spec/operators/delay-spec.js [hunk 1]
```diff
@@ -1,11 +1,79 @@
...
+
+  it('should raise error when source raises error by absolute time period', function () {
+    var e1 =   hot('--a--b--#');
+    var expected = '-----a--b--#';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
+
+    expectObservable(e1.delay(absoluteDelay, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should raise error when source raises error by absolute time period after subscription', function () {
```
Comment (kwonoj) [inline spec/operators/delay-spec.js]: This behavior is bit different to RxJS4 [test case] (https://github.com/Reactive-Extensions/RxJS/blob/master/tests/observable/delay.js#L156-L256), in RxJS4 test case is described as

```
source  = '---a---b---#'
delayed = '----a----b-#'; 
delayed = hot(source).delay(10);
```

that `error` is emitted as soon as source emits. Per [documantation](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/delay.md), behavior is defined as 

`Time shifts the observable sequence by dueTime. The relative time intervals between the values are preserved.`
think `error` should be considered as same as normal value element, but maybe that's something I understood behavior incorrectly(or either read RxJS4 test cases incorrectly?), asking confirmation. @blesh @staltz

File: spec/operators/delay-spec.js [hunk 2]
```diff
@@ -1,11 +1,79 @@
...
+    var e1 =   hot('---^---a---b---#');
+    var expected =    '-------a---b---#';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
```
Comment (kwonoj) [inline spec/operators/delay-spec.js]: This behavior is bit different to RxJS4 [test case] (https://github.com/Reactive-Extensions/RxJS/blob/master/tests/observable/delay.js#L156-L256), in RxJS4 test case is described as

```
source  = '---a---b---#'
delayed = '----a----b-#'; 
delayed = hot(source).delay(10);
```

that `error` is emitted as soon as source emits. Per [documantation](https://github.com/Reactive-Extensions/RxJS/blob/master/doc/api/core/operators/delay.md), behavior is defined as 

`Time shifts the observable sequence by dueTime. The relative time intervals between the values are preserved.`
think `error` should be considered as same as normal value element, but maybe that's something I understood behavior incorrectly(or either read RxJS4 test cases incorrectly?), asking confirmation. @blesh @staltz

Comment (staltz) [inline spec/operators/delay-spec.js]: IMO errors should be reported ASAP because they mean "unhandled exceptions".

Comment (kwonoj) [inline spec/operators/delay-spec.js]: OK, then I can consider this is my-incorrect-understanding, right?

File: spec/operators/delay-spec.js [hunk 3]
```diff
@@ -1,11 +1,79 @@
...
+    var e1 =   hot('---^--a--b--|');
+    var expected =    '------a--b--|';
+    var absoluteDelay = new Date(rxTestScheduler.now() + 30);
+
+    expectObservable(e1.delay(absoluteDelay, rxTestScheduler)).toBe(expected);
+  });
+
+  it('should raise error when source raises error', function () {
+    var e1 =   hot('---a---b---#');
+    var expected = '------a---b#';
```
Comment (kwonoj) [inline spec/operators/delay-spec.js]: quickly mended `error` behavior per suggested. @staltz , whip me if I did something incorrect still.

Comment (staltz) [inline spec/operators/delay-spec.js]: Since you're at it, how about adding subscription assertions too?

Comment (kwonoj) [inline spec/operators/delay-spec.js]: subscription for sources, am I understanding correct? i.e for this testcase,

```
var e1 =   hot('---a---b---#');
var e1Sub =    '           !';
....
```

Comment (staltz) [inline spec/operators/delay-spec.js]: `var e1Sub = '^         !';` it was missing `^` to indicate when it's subscribed.

Comment (kwonoj) [inline spec/operators/delay-spec.js]: Yeah, too hesitate to write down stubs. Thanks for suggestion, trying to update it.

**Comment by blesh** [comment]
Looks good.

**Comment by blesh** [comment]
merged with b109100... thanks @kwonoj

**Pull request closed without merging**